### ISSUE TITLE
Dead End Aegis (1835830) and Dead End Aegis: Gaiden (1839960)

### ISSUE THREAD

**Issue by fallenguru** [ValveSoftware/Proton#6082]
# Compatibility Report
Dead End Aegis (1835830) / Dead End Aegis Gaiden (1839960) – English versions

The first is the main game, the second is a prequel side story. They both use exactly the same engine (only with different assets), which is why I grouped them into one report. That, and DEA Gaiden, being free and SFW, is much easier to test.

## System Information
- GPU: AMD Radeon VII
- Driver/LLVM version: Mesa 22.1.5\~kisak1\~b [Kisak Mesa PPA]
- Kernel version: 5.4.0-120-lowlatency [Ubuntu 18.04.6 LTS]
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/fallenguru/116961c075c5d5048a7ef28a313b9421
- Proton versions: any recent WINE version (tried Proton 7.0-3 and Experimental, GE 7.29, vanilla WINE 7.1*x*)

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

[steam-1839960.log](https://github.com/ValveSoftware/Proton/files/9292994/steam-1839960.log)

## Symptoms 

The text rendering is broken. The text is too large, so it gets slightly cut off at the top and massively to the right. It doesn't remotely fit the textbox and is therefore **completely unplayable**. If you look at screenshots ([Steam store page](https://store.steampowered.com/app/1839960/), for example), the font is wrong to begin with.

Screenshot [taken with Proton 7.03]:

![DEA Gaiden text rendering issue](https://user-images.githubusercontent.com/16985028/183711410-0027360c-3c41-47fb-889c-f5e2eb5d5d73.png)

That's despite the fact that it ships with the font (Neuton.ttf). I tried adding it to the fontconfig path for good measure. Fontconfig can see it and I can use it just fine, but as for the game, no dice. 
Hypothesis: Something in the way the font is requested causes the font selection process to disqualify it and fall through to an incompatible fallback.

The issue does not seem to be system-dependent, I've had it on multiple systems, there are matching ProtonDB entries ([main game](https://www.protondb.com/app/1835830), [Gaiden](https://www.protondb.com/app/1839960)), and at least one report on Reddit.

## Reproduction

Start game, start new game. If desired, click to advance. Notice that all the lines are cut off (most are much longer than what's shown).

**Comment by juliodion12345** [comment]
![Screenshot_20220809_194039](https://user-images.githubusercontent.com/5234947/183791827-5f450af1-6bc2-4f9d-acb8-2318a9eaf2fb.png)

I also noted that the sample text in the Settings -> Screen/Text is cut

The complete text is this one: Please　set　text　speed　to　the　speed　of　your　choosing.

**Comment by fallenguru** [comment]
Makes sense. It's just a sample textbox so you can see immediately what the settings (speed, opacity) do. Might be even quicker to enter settings than start a new game to test, though. Good find.

**Comment by fallenguru** [comment]
Adding a DLL override for `d2d1.dll` (so it uses the one it ships with) fixes the font issue.

Can this tweak be preconfigured Proton-side, either by Valve or by the developer?

**Comment by juliodion12345** [comment]
> Adding a DLL override for `d2d1.dll` (so it uses the one it ships with) fixes the font issue.
> 
> Can this tweak be preconfigured Proton-side, either by Valve or by the developer?

I can confirm that this works

![Screenshot_20220905_230039](https://user-images.githubusercontent.com/5234947/188550632-85581bb2-9305-45e5-b5b4-e2b719f5ddb1.png)

**Comment by Saetron** [comment]
> Adding a DLL override for `d2d1.dll` (so it uses the one it ships with) fixes the font issue.
> 
> Can this tweak be preconfigured Proton-side, either by Valve or by the developer?

Can confirm that adding d2d1 with winecfg fixes this issue for DEA: Gaiden and DEA

**Comment by MrMPN** [comment]
While the fix does work, it seems to me that there's still somewhat of a problem. The font looks correct now, but it still kinda goes a bit beyond the area it should be. You can only see it on long sentences. It's not unreadable as it does not go past the screen, but it is inconvenient. 

It feels like the area the text fits into is "read" incorrectly if that makes sense, so maybe it's something related to the visual novel engine.

**Comment by Saetron** [comment]
> While the fix does work, it seems to me that there's still somewhat of a problem. The font looks correct now, but it still kinda goes a bit beyond the area it should be. You can only see it on long sentences. It's not unreadable as it does not go past the screen, but it is inconvenient.
> 
> It feels like the area the text fits into is "read" incorrectly if that makes sense, so maybe it's something related to the visual novel engine.

I just checked Sadistic Blood (same engine, same issue and same fix) and it looks like that text overflow is also present on Windows, so its an issue with the engine, nothing with proton.
